smt/pkg/smt/smt_batch.go

```diff
@@ -143,27 +137,52 @@ func (s *SMT) InsertBatch(ctx context.Context, logPrefix string, nodeKeys []*uti
 			maxInsertingNodePathLevel = insertingNodePathLevel
 		}
 	}
+	progressChan <- uint64(1)
+	stopProgressPrinter()
 
 	s.updateDepth(maxInsertingNodePathLevel)
 
+	totalDeleteOps := 0
```

Comment:
Three nested loops are not good for performance especially when it is just for logs. Do you think we could find another solution?

Reply:
I think only the top level is enough tbh